tests/ifsbench/ifsbench_run.py

```diff
@@ -0,0 +1,366 @@
+#! @IFSBENCH_PYTHON@
+
+# (C) Copyright 2024- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from contextlib import nullcontext
+from pathlib import Path
+import re
+from tempfile import TemporaryDirectory
+from typing import List, Dict
+
+import click
+import yaml
+
+from ifsbench import (
+    Benchmark,
+    cli,
+    debug,
+    DataFileStats,
+    DefaultApplication,
+    DefaultArch,
+    EnvHandler,
+    error,
+    Job,
+    PydanticConfigMixin,
+    PydanticDataFrame,
+    ScienceSetup,
+    TechSetup
+)
+
+from ifsbench.data import FetchHandler, NamelistHandler, NamelistOverride, RenameHandler, RenameMode
+from ifsbench.validation import FrameCloseValidation
+
+class EclandResult(PydanticConfigMixin):
```

Comment:
[no action] The `PydanticConfigMixin` is in general an internal implementation detail, which should not be exposed to user-facing code. In later versions, maybe something like `ifsbench.Result` could be introduced as a base class, so that this becomes `class EclandResult(Result)`. However, this is not needed for this initial version, just a comment for future development.